src/coreclr/vm/interpexec.h

```diff
@@ -27,21 +27,84 @@ struct StackVal
 typedef DPTR(struct InterpMethodContextFrame) PTR_InterpMethodContextFrame;
 class InterpreterFrame;
 
+enum class InterpreterFrameReporting
+{
+    Normal = 0,
+#ifndef FEATURE_REUSE_INTERPRETER_STACK_FOR_NORMAL_FUNCLETS
+    FuncletReportGlobals = 1,
+    FuncletNoReportGlobals = 2,
+
+    Mask = 3
+#endif // FEATURE_REUSE_INTERPRETER_STACK_FOR_NORMAL_FUNCLETS
+};
+
 struct InterpMethodContextFrame
 {
-    PTR_InterpMethodContextFrame pParent;
-    PTR_InterpByteCodeStart startIp; // from startIp we can obtain InterpMethod and MethodDesc
-    int8_t *pStack;
-    int8_t *pRetVal;
-    const int32_t *ip; // This ip is updated only when execution can leave the frame
-    PTR_InterpMethodContextFrame pNext;
+    PTR_InterpMethodContextFrame pParent = 0;
+    PTR_InterpByteCodeStart startIp = 0; // from startIp we can obtain InterpMethod and MethodDesc
+    int8_t *pStack = 0;
+private:
+    int8_t *pRetVal = 0; // If the low bit is set on pRetVal, then Frame represents a funclet
```

Comment:
This comment mentions 'the low bit' but it looks like we actually reserve two bits based on Mask above